WooCommerce/Classes/POS/Models/PointOfSaleAggregateModel.swift

```diff
@@ -201,14 +199,22 @@ extension PointOfSaleAggregateModel {
         }
     }
 
-    func startCashPayment() {
-        isCashPaymentInProgress = true
+    // Prevents card payments from the moment the merchant decides we start collecting cash
+    // Once we get the callback from the card service, we switch to cash collection state
+    @MainActor
+    func startCashPayment() async {
+        try? await cardPresentPaymentService.cancelPayment()
```

Comment:
Is it worth handling the errors here? I think it probably is, but not totally sure how. Starting the cash payment doesn't seem like the way though...

Feel free to do this under a new issue.

Reply:
Logged: https://github.com/woocommerce/woocommerce-ios/issues/14858